### ISSUE TITLE
MA0007 Should not show when using or keyword with inline switch

### ISSUE THREAD

**Issue by Tyme-Bleyaert** [meziantou/Meziantou.Analyzer#806]
### Version of the Meziantou.Analyzer NuGet package

2.0.201

### Rule Identifier

MA0007

### Target Framework

.NET 8

### C# Language version

C# 12

### Description

Issue: 
Warning MA0007 shown when using or keyword in inline switch statement.

### Reproduction Steps

Minimal code:

```c#
public enum Status {
    New = 0,
    InProgress,
    Resolved,
}

internal static string GetTranslation(this Status status)
{
    return status switch
    {
        Status.InProgress => "In progress",
        Status.Resolved => "Fixed",
        Status.New or _ => "New" // MA0007 here
    };
}
```

**Comment by Tyme-Bleyaert** [comment]
Nvm, im dumb.
